android-json-form-wizard/src/main/java/com/vijay/jsonwizard/task/AttachRepeatingGroupTask.java

```diff
@@ -100,14 +104,27 @@ protected void onPostExecute(List<View> result) {
                     keysToRemove.add(repeatingGroupKey);
                     rootLayout.removeViewAt(i);
                 }
-                // remove deleted fields from form json
+//                remove deleted fields from form json
+                ArrayList<String> removeThisFields = new ArrayList<>();
                 int len = fields.length();
                 for (int i = len - 1; i >= 0; i--) {
                     String[] key = ((String) fields.getJSONObject(i).get(KEY)).split("_");
                     if (keysToRemove.contains(key[key.length - 1])) {
+                        removeThisFields.add((String) fields.getJSONObject(i).get(KEY));
                         fields.remove(i);
                     }
                 }
+//                remove deleted views from form json
+                Collection<View> viewCollection =  widgetArgs.getFormFragment().getJsonApi().getFormDataViews();
```

Comment:
I think you should check for null in this chained method calls `widgetArgs.getFormFragment().getJsonApi().getFormDataViews()`